url354ce1a2d1_dagaren_gladiator_chess_tgz-pJ8-es_dagaren_gladiator_main_MainJ8/src/es/dagaren/gladiator/representation/BitboardUtils.java

```diff
@@ -0,0 +1,216 @@
+      bitboard ^= a ^ (a >>> 14);
+      a = b  & (bitboard ^ (bitboard << 7));
+      bitboard ^= a ^ (a >>> 7);
+      
+      return bitboard;
+   }
+   
+   /**
+    * Vuncion que devuelve un bitboard rotado 90 grados
+    * en sentido antihorario, tal como se muestra en la imagen:
+    * 
+    * . 1 1 1 1 . . .     . . . . . . . .
+    * . 1 . . . 1 . .     . . . . . . . .
+    * . 1 . . . 1 . .     . 1 1 . . . . 1
+    * . 1 . . 1 . . .     1 . . 1 . . 1 .
+    * . 1 1 1 . . . .     1 . . . 1 1 . .
+    * . 1 . 1 . . . .     1 . . . 1 . . .
+    * . 1 . . 1 . . .     1 1 1 1 1 1 1 1
+    * . 1 . . . 1 . .     . . . . . . . .
+    * 
+    * Para realizar la rotación se realiza primero un volteo vertical
+    * del bitboard, y al resultado se le vuelve a realizar un volteo
+    * alrededor de la diagonal principal (a1a8)
+    *  
+    * @param bitboard
+    * @return
+    */
+   public static long rotate90DegreesLeft(long bitboard){
+      return flipDiagonalA1H8(flipVertical(bitboard));
+   }
+   
+   /**
+    * Función que devuelve un bitboard rotado 90 grados
+    * en sentido horario, tal como se muestra en la imagen:
+    * 
+    * . 1 1 1 1 . . .     . . . . . . . .
+    * . 1 . . . 1 . .     1 1 1 1 1 1 1 1
+    * . 1 . . . 1 . .     . . . 1 . . . 1
+    * . 1 . . 1 . . .     . . 1 1 . . . 1
+    * . 1 1 1 . . . .     . 1 . . 1 . . 1
+    * . 1 . 1 . . . .     1 . . . . 1 1 .
+    * . 1 . . 1 . . .     . . . . . . . .
+    * . 1 . . . 1 . .     . . . . . . . .
+    * 
+    * Para realizar la rotación se realiza primero un volteo alrededor
+    * de la diagonal principal (a1a8) y al resultado se le vuelve
+    * a realizar un volteo vertical
+    *  
+    * @param bitboard El bitboard que se desea rotar
+    * @return
+    */
+   public static long rotate90DegreesRight(long bitboard){
+      return flipVertical(flipDiagonalA1H8(bitboard));
+      
+   }
+   
+   /**
+    * Función que realiza una pseudorotación de 45 grados en sentido
+    * horario. El concepto de rotación de 45 grados es el que se muestra
+    * en la siguiente imagen, en la que se númeran los bits de la diagonal
+    * principal del 0 a F:
+    * 
+    * 9 A B C D E F 0     9 1 1 1 1 1 1 1
+    * A B C D E F 0 1     A A 2 2 2 2 2 2
+    * B C D E F 0 1 2     B B B 3 3 3 3 3
+    * C D E F 0 1 2 3     C C C C 4 4 4 4
+    * D E F 0 1 2 3 4     D D D D D 5 5 5
+    * E F 0 1 2 3 4 5     E E E E E E 6 6
+    * F 0 1 2 3 4 5 6     F F F F F F F 7
+    * 0 1 2 3 4 5 6 7     0 0 0 0 0 0 0 0
+    * 
+    * @param bitboard
+    * @return
+    */
+   public static long rotate45DegreesRight(long bitboard){
+      long k1 = 0xAAAAAAAAAAAAAAAAL;
+      long k2 = 0xCCCCCCCCCCCCCCCCL;
+      long k4 = 0xF0F0F0F0F0F0F0F0L;
+      bitboard ^= k1 & (bitboard ^ Long.rotateRight(bitboard, 8));
+      bitboard ^= k2 & (bitboard ^ Long.rotateRight(bitboard, 16));
+      bitboard ^= k4 & (bitboard ^ Long.rotateRight(bitboard, 32));
+      return bitboard;
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 116:122, 126:134, 156:162, 166:173.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.